pkg/collector/upgrade/v0_122_0_test.go

```diff
@@ -0,0 +1,341 @@
+					Name:      "test-collector",
+					Namespace: "default",
+				},
+				Spec: v1beta1.OpenTelemetryCollectorSpec{
+					Config: v1beta1.Config{
+						Service: v1beta1.Service{
+							Telemetry: &v1beta1.AnyConfig{
+								Object: map[string]interface{}{
+									"metrics": map[string]interface{}{
+										"address": "",
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "should not modify config when prometheus reader is already configured",
+			initialConfig: &v1beta1.OpenTelemetryCollector{
+				TypeMeta: metav1.TypeMeta{
+					Kind:       "OpenTelemetryCollector",
+					APIVersion: "v1beta1",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test-collector",
+					Namespace: "default",
+				},
+				Spec: v1beta1.OpenTelemetryCollectorSpec{
+					Config: v1beta1.Config{
+						Service: v1beta1.Service{
+							Telemetry: &v1beta1.AnyConfig{
+								Object: map[string]interface{}{
+									"metrics": map[string]interface{}{
+										"readers": []interface{}{
+											map[string]interface{}{
+												"pull": map[string]interface{}{
+													"exporter": map[string]interface{}{
+														"prometheus": map[string]interface{}{
+															"host": "0.0.0.0",
+															"port": int32(8888),
+														},
+													},
+												},
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			expectedConfig: &v1beta1.OpenTelemetryCollector{
+				TypeMeta: metav1.TypeMeta{
+					Kind:       "OpenTelemetryCollector",
+					APIVersion: "v1beta1",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test-collector",
+					Namespace: "default",
+				},
+				Spec: v1beta1.OpenTelemetryCollectorSpec{
+					Config: v1beta1.Config{
+						Service: v1beta1.Service{
+							Telemetry: &v1beta1.AnyConfig{
+								Object: map[string]interface{}{
+									"metrics": map[string]interface{}{
+										"readers": []interface{}{
+											map[string]interface{}{
+												"pull": map[string]interface{}{
+													"exporter": map[string]interface{}{
+														"prometheus": map[string]interface{}{
+															"host": "0.0.0.0",
+															"port": int32(8888),
+														},
+													},
+												},
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "should not modify config when OTLP is configured",
+			initialConfig: &v1beta1.OpenTelemetryCollector{
+				TypeMeta: metav1.TypeMeta{
+					Kind:       "OpenTelemetryCollector",
+					APIVersion: "v1beta1",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test-collector",
+					Namespace: "default",
+				},
+				Spec: v1beta1.OpenTelemetryCollectorSpec{
+					Config: v1beta1.Config{
+						Service: v1beta1.Service{
+							Telemetry: &v1beta1.AnyConfig{
+								Object: map[string]interface{}{
+									"metrics": map[string]interface{}{
```

Comment:
OTLP looks a bit different:
```
          readers:
            - periodic:
                interval: 30000
                exporter:
                  otlp:
                    protocol: http/protobuf
                    endpoint: http://${env:MY_POD_IP}:4318
```

Reply:
You are right. I copied/pasted and later not commit. Sorry again.

Reply:
np np! thanks for all the diligence here.